pkg/testing/tools/artifacts_api.go

```diff
@@ -0,0 +1,246 @@
+	Dependencies                 []any              `json:"dependencies"`
+}
+
+type BuildDetails struct {
+	Build struct {
+		Projects             map[string]Project `json:"projects"`
+		StartTime            string             `json:"start_time"`
+		ReleaseBranch        string             `json:"release_branch"`
+		Prefix               string             `json:"prefix"`
+		EndTime              string             `json:"end_time"`
+		ManifestVersion      string             `json:"manifest_version"`
+		Version              string             `json:"version"`
+		Branch               string             `json:"branch"`
+		BuildID              string             `json:"build_id"`
+		BuildDurationSeconds int                `json:"build_duration_seconds"`
+	} `json:"build"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type SearchPackageResult struct {
+	Packages  map[string]Package `json:"packages"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type httpDoer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type ArtifactAPIClient struct {
+	c      httpDoer
+	apiURL string
+}
+
+func (aac ArtifactAPIClient) GetVersions(ctx context.Context) (list *VersionList, err error) {
+	joinedURL, err := url.JoinPath(aac.apiURL, artifactsAPIV1VersionsEndpoint)
+	if err != nil {
+		err = fmt.Errorf("composing URL with %q %q: %w", aac.apiURL, artifactsAPIV1VersionsEndpoint, err)
+		return
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, joinedURL, nil)
+
+	if err != nil {
+		err = fmt.Errorf("composing request for getting versions: %w", err)
+		return
+	}
+
+	resp, err := aac.c.Do(req)
+	if err != nil {
+		err = fmt.Errorf("executing http request %v: %w", req, err)
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		err = fmt.Errorf("%d: %w", resp.StatusCode, ErrBadHTTPStatusCode)
+		return
+	}
+
+	respBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		err = fmt.Errorf("reading response body: %w", err)
+		return
+	}
+	list = new(VersionList)
+	err = json.Unmarshal(respBytes, list)
+
+	if err != nil {
+		err = fmt.Errorf("unmarshaling version list: %w", err)
+		return
+	}
+
+	return list, nil
+}
+
+func (aac ArtifactAPIClient) GetBuildsForVersion(ctx context.Context, version string) (builds *VersionBuilds, err error) {
+	joinedURL, err := url.JoinPath(aac.apiURL, fmt.Sprintf(artifactsAPIV1VersionBuildsEndpoint, version))
+	if err != nil {
+		err = fmt.Errorf("composing URL with %q %q: %w", aac.apiURL, fmt.Sprintf(artifactsAPIV1VersionBuildsEndpoint, version), err)
+		return
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, joinedURL, nil)
+
+	if err != nil {
+		err = fmt.Errorf("composing request for getting versions: %w", err)
```

Comment:
[Suggestion]
_composing_ to me sound as if you're getting 2 pieces of requests and make 1 out of them.
If i'd see this log I'd not guess it came for `http.NewRequestWithContext`, I'd think it was some custom code dealing with a request.

```suggestion
		err = fmt.Errorf("creating request for getting versions: %w", err)
```